### ISSUE TITLE
better-typescript-lib@2.2.0 does not work with @typescript-eslint/no-throw-literal

### ISSUE THREAD

**Issue by kasugaiasuka** [uhyo/better-typescript-lib#18]
**Steps to reproduce:**

1. Install `better-typescript-lib@2.2.0`.
2. Install `@typescript-eslint/eslint-plugin@5.46.1` and other necessities.
3. Activate `@typescript-eslint/no-throw-literal` rule.
4. Lint the code below.

```typescript
throw new Error("foo");
```

**Expected:**

Nothing is flagged.

**Actual:**

```
error  Expected an error object to be thrown  @typescript-eslint/no-throw-literal
```

**Additional information:**

`better-typescript-lib@2.1.0` works well.

**Comment by kasugaiasuka** [comment]
I found this probrem is already reported at https://github.com/typescript-eslint/typescript-eslint/issues/6207

**Comment by uhyo** [comment]
As to why this didn't happen. on 2.1.0, that version didn't replace _all_  default libs that defined `Error`. [This check](https://github.com/typescript-eslint/typescript-eslint/blob/c3cfe22b4ac152de42a136678dfda50d766a59c8/packages/eslint-plugin/src/rules/no-throw-literal.ts#L69-L77) pass if there is at least one declaration of `Error` that points to TypeScript's built-in lib.

I think the root cause is TypeScript's `isSourceFileDefaultLibrary` behavior, so I will try reporting an issue for that.

**Comment by uhyo** [comment]
Fixed in `2.2.1`.

This release added `/// <reference no-default-lib="true"/>` to all library files and now typescript-eslint works fine.
